docs/docs/advanced/deploy-eliza-to-tee-with-oasis-rofl.md

```diff
@@ -0,0 +1,241 @@
+
+- [Oasis CLI](https://github.com/oasisprotocol/cli/releases) installed
+- Docker Desktop or Orbstack (Orbstack is recommended)
+- Testnet tokens from the [Oasis Faucet](https://faucet.testnet.oasis.dev/)
+- A character file for your Eliza agent (you can use the example character file or create your own)
+
+---
+
+### Environment Setup
+
+To set up your environment for TEE development:
+
+1. **Create an Oasis Account**
+
+    ```bash
+    oasis-cli account create --name <account-name>
+    ```
+
+2. **Request Testnet Tokens**
+
+    Visit the [Oasis Faucet](https://faucet.testnet.oasis.dev/) to request testnet tokens.
+
+3. **Create a ROFL Application**
+
+    Create the ROFL application:
+
+    ```bash
+    oasis rofl create --network testnet --account <account-name>
+    ```
+
+### Build and Deploy the ROFL app
+
+1. **Configure the ROFL app**
+
+    Update the `rofl-compose.yaml` file with your configuration:
+
+    ```yaml
+    services:
+      eliza:
+        restart: always
+        image: ghcr.io/elizaos/eliza:latest
+        build:
+          context: .
+          dockerfile: Dockerfile
+        platform: linux/amd64
+        environment:
+          - CHARACTERS=${CHARACTERS}
+          - SERVER_PORT=${SERVER_PORT}
+          # Add other required environment variables
+    ```
+
+2. **Encrypt Secrets**
+
+    Encrypt all required environment variables. You can use any character file of your choice - here we're using the staff engineer character as an example. Refer to the Secret Management section above for a comprehensive list of secrets you might want to set:
+
+    ```bash
+    # Set the character file path (you can use any character file of your choice)
+    echo -n "characters/staff-engineer.character.json" | oasis rofl secret set CHARACTERS -
+    echo -n "3000" | oasis rofl secret set SERVER_PORT -
+    # Add other required secrets
+    ```
+
+3. **Build the ROFL Application**
+
+    For MacOS users, use the following Docker image:
+
+    ```bash
+    docker run --platform linux/amd64 --volume .:/src --rm -it ghcr.io/oasisprotocol/rofl-dev:main
+    ```
+
+    Inside the container, build the ROFL application:
+
+    ```bash
+    oasis rofl build --update-manifest
+    ```
+
+4. **Deploy the ROFL Application**
+
+    You have two options for deployment:
+
+    A. **Run Your Own Oasis Node**
+
+    1. Follow the [Oasis Node Setup Guide](https://docs.oasis.io/node/run-your-node/paratime-client-node)
+    2. Copy the ROFL bundle to your node
+    3. Update your node configuration
+    4. Restart your node
+
+    B. **Deploy to Oasis Provider**
+
+    1. Upload the ROFL bundle to a public file server
+    2. Contact the Oasis team on [Discord](https://oasis.io/discord) #dev-central channel
+
+### Verify TEE Deployment
+
+Once deployed, you can verify the TEE deployment by:
+
+1. Checking the ROFL application status:
+
+    ```bash
+    oasis rofl show
+    ```
+
+2. Monitoring the application logs
+3. Verifying remote attestation
```

Comment:
Part of the Oasis CLI PR above is also this.

````suggestion
Once deployed, you can verify whether your ROFL instance is running, the latest logs and the runtime-attestation key:

    ```bash
    oasis rofl machine show
    ```
````